upload_categories.py

```diff
@@ -0,0 +1,201 @@
+import firebase_admin
+from firebase_admin import credentials, firestore
+from dataclasses import dataclass
+from typing import List
+
+@dataclass
+class Subcategory:
+    id: str
+    name: str
+    tags: List[str]
+
+@dataclass
+class Category:
+    id: str
+    name: str
+    description: str
+    subcategories: List[Subcategory]
+
+def initialize_firestore():
+    # Initialize the Firebase app
+    cred = credentials.Certificate('data_loader_service_account_key.json')
+    firebase_admin.initialize_app(cred)
+    return firestore.client()
+
+def upload_data(db, categories: List[Category]):
+    for category in categories:
+        # Create or update the category document
+        category_ref = db.collection('categories').document(category.id)
+        category_data = {
+            'name': category.name,
+            'description': category.description
+        }
+        category_ref.set(category_data)
+
+        # Upload subcategories
+        for subcategory in category.subcategories:
+            subcategory_ref = category_ref.collection('subcategories').document(subcategory.id)
+            subcategory_data = {
+                'name': subcategory.name,
+                'tags': subcategory.tags
+            }
+            subcategory_ref.set(subcategory_data)
+
+    print("Data uploaded successfully.")
+
+def main():
+    db = initialize_firestore()
+
+    # Define your categories, subcategories, and tags
+    categories = [
```

Comment:
Great initiative! Now, everyone can upload their new categories quickly without spending hours doing it manually. Fantastic job!